gslib/tests/test_rpo.py

```diff
@@ -0,0 +1,186 @@
+
+from gslib.commands.rpo import RpoCommand
+from gslib.exception import CommandException
+from gslib.gcs_json_api import GcsJsonApi
+from gslib.storage_url import StorageUrlFromString
+import gslib.tests.testcase as testcase
+from gslib.tests.testcase.integration_testcase import SkipForGS
+from gslib.tests.testcase.integration_testcase import SkipForJSON
+from gslib.tests.testcase.integration_testcase import SkipForXML
+from gslib.tests.util import ObjectToURI as suri
+from gslib.tests.util import SetBotoConfigForTest
+
+from six import add_move, MovedModule
+
+add_move(MovedModule('mock', 'mock', 'unittest.mock'))
+from six.moves import mock
+
+
+class TestRpoUnit(testcase.GsUtilUnitTestCase):
+
+  def test_get_for_multiple_bucket_calls_api(self):
+    bucket_uri1 = self.CreateBucket(bucket_name='rpofoo')
+    bucket_uri2 = self.CreateBucket(bucket_name='rpobar')
+    stdout = self.RunCommand(
+        'rpo',
+        ['get', suri(bucket_uri1), suri(bucket_uri2)],
+        return_stdout=True)
+    expected_string = textwrap.dedent("""\
+      gs://rpofoo: None
+      gs://rpobar: None
+      """)
+    self.assertEqual(expected_string, stdout)
+
+  def test_get_with_wildcard(self):
+    self.CreateBucket(bucket_name='boo1')
+    self.CreateBucket(bucket_name='boo2')
+    stdout = self.RunCommand('rpo', ['get', 'gs://boo*'], return_stdout=True)
+    actual = '\n'.join(sorted(stdout.strip().split('\n')))
+    expected_string = textwrap.dedent("""\
+      gs://boo1: None
+      gs://boo2: None""")
+    self.assertEqual(actual, expected_string)
+
+  def test_get_with_wrong_url_raises_error(self):
+    with self.assertRaisesRegexp(CommandException, 'No URLs matched'):
+      self.RunCommand('rpo', ['get', 'gs://invalid*'])
+
+  def test_set_called_with_incorrect_value_raises_error(self):
+    with self.assertRaisesRegexp(
+        CommandException,
+        r'Invalid value for rpo set. Should be one of \(ASYNC_TURBO\|DEFAULT\)'
+    ):
+      self.RunCommand('rpo', ['set', 'random', 'gs://boo*'])
+
+  def test_invalid_subcommand_raises_error(self):
+    with self.assertRaisesRegexp(
+        CommandException, 'Invalid subcommand "blah", use get|set instead'):
+      self.RunCommand('rpo', ['blah', 'DEFAULT', 'gs://boo*'])
+
+
+class TestRpo(testcase.GsUtilIntegrationTestCase):
+  """Integration tests for rpo command."""
+
+  # TODO: Delete this method once rpo get results are consistent
+  # from the backend, and replace this call with
+  # VerifyCommandGet(bucket_uri, 'rpo', 'DEFAULT').
+  # Currently, the rpo results are inconsistent
+  # and None is a valid default value for all the buckets.
+  # See b/197251750#comment19
+  def _verify_get_returns_default_or_none(self, bucket_uri):
+    """Checks if the rpo get command returns default."""
+    try:
+      self.VerifyCommandGet(bucket_uri, 'rpo', 'DEFAULT')
+    except AssertionError:
+      self.VerifyCommandGet(bucket_uri, 'rpo', 'None')
+
+  @SkipForXML('RPO only runs on GCS JSON API')
```

Comment:
Periods at the end of the decorator messages

Reply:
Done.